app/pages/multisig/MultiSignatureCreateProposal/BuildProposal.tsx

```diff
@@ -57,7 +57,7 @@ export default function BuildProposal({
         );
 
         if (newProposal) {
-            onFinish(newProposal, fields);
+            onFinish(newProposal as MultiSignatureTransaction, fields);
```

Comment:
calling something like an `onFinish` event with a partial transaction doesn't make much sense, when the components purpose is to build a full transaction. Otherwise we'd need a generic way to ensure all fields are set on the transaction. I don't remember why it wasn't possible to change the return type of `handler.createTransaction` to not be a `Partial` type though.

Reply:
I believe its because it does not have the transaction id at that point? which is why it probably shouldn't be cast like this?